applications/fishing-map/src/features/download/DownloadTrackModal.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { event as uaEvent } from 'react-ga'
+import { useDispatch, useSelector } from 'react-redux'
+import Modal from '@globalfishingwatch/ui-components/dist/modal'
+import { Button, Choice, Icon, Tag } from '@globalfishingwatch/ui-components/dist'
+import {
+  DownloadTrackParams,
+  selectDownloadTrackStatus,
+  selectDownloadTrackId,
+  selectDownloadTrackName,
+  downloadTrackThunk,
+  resetDownloadTrackStatus,
+  selectDownloadTrackDataset,
+  clearDownloadTrackVessel,
+} from 'features/download/downloadTrack.slice'
+import { EMPTY_FIELD_PLACEHOLDER } from 'utils/info'
+import { TimelineDatesRange } from 'features/map/controls/MapInfo'
+import { useTimerangeConnect } from 'features/timebar/timebar.hooks'
+import { DateRange } from 'features/analysis/analysis.slice'
+import { AsyncReducerStatus } from 'utils/async-slice'
+import styles from './DownloadModal.module.css'
+import { Format, FORMAT_OPTIONS } from './downloadTrack.config'
+
+function DownloadTrackModal() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const timeoutRef = useRef<NodeJS.Timeout>()
+  const downloadStatus = useSelector(selectDownloadTrackStatus)
+  const [format, setFormat] = useState(FORMAT_OPTIONS[0].id as Format)
+  const { timerange } = useTimerangeConnect()
+
+  const downloadTrackId = useSelector(selectDownloadTrackId)
+  const downloadTrackName = useSelector(selectDownloadTrackName)
+  const downloadTrackDataset = useSelector(selectDownloadTrackDataset)
+
+  const onDownloadClick = async () => {
+    const downloadParams: DownloadTrackParams = {
+      vesselId: downloadTrackId,
+      vesselName: downloadTrackName,
+      dateRange: timerange as DateRange,
+      datasets: downloadTrackDataset,
+      format,
+    }
+
+    try {
+      await dispatch(downloadTrackThunk(downloadParams))
```

Comment:
do we want to keep the modal open ? otherwise this would be the place to auto-close it